packages/core/src/network.js

```diff
@@ -194,6 +194,14 @@ export class Network {
     // do not handle data urls
     if (request.url.startsWith('data:')) return;
 
+    // Browsers handle URL encoding leniently, but invalid characters can break tools like Jackproxy.
+    // This code checks for issues such as `%` and leading spaces and warns the user accordingly.
+    decodeAndEncodeURLWithLogging(request.url, this.log, {
+      meta: { ...this.meta, url: request.url },
+      shouldLogWarning: true,
```

Comment:
`shouldLogWarning: true`, this is always true so do we even need this as an argument?

Reply:
- `shouldLogWarning: true`, in our case Yes, but this function can be used in other cases as well in future to only get modified url and don't want to show the warning, so made it a little generic to use.